code/network/multimsgs.cpp

```diff
@@ -5632,8 +5684,9 @@ void send_player_settings_packet(net_player *p)
 			// break the p_info structure by member, so we don't overwrite any absolute pointers
 			// ADD_DATA(Net_players[idx].p_info);
 			ADD_INT(Net_players[idx].p_info.team);
-			ADD_INT(Net_players[idx].p_info.ship_index);
-			ADD_INT(Net_players[idx].p_info.ship_class);
+			ADD_SHORT(static_cast<short>(Net_players[idx].p_info.ship_index));
+			val = static_cast<char>(Net_players[idx].p_info.ship_class);
```

Comment:
Hold it, this is backwards from before.  ship_index was a char while ship_class was a short.

Reply:
Oof, good catch. That is indeed backwards.